tests/async/test_character_materialization_readiness.py

```
@@ -0,0 +1,235 @@
+
+    def test_secure_destination_rejects_unsafe_or_existing_paths(self):
+        """Reject every unsafe receipt destination before artifact creation."""
+        with tempfile.TemporaryDirectory() as temporary:
+            root = Path(temporary)
+            protected = root / "protected"
+            protected.mkdir(mode=0o700)
+            existing = protected / "existing.receipt"
+            existing.write_bytes(b"existing")
+            open_parent = root / "open"
+            open_parent.mkdir(mode=0o755)
+            os.chmod(open_parent, 0o755)
```

Ruff (0.16.3)

[error] `os.chmod` setting a permissive mask `0o755` on file or directory

(S103)